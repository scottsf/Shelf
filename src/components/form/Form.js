import React, {Component} from 'react';
import './form.scss';
import axios from 'axios';

class Form extends Component {
  constructor() {
    super();
    this.state = {
      img: '',
      name: '',
      price: '',
      id: null,
    }
    this.saveProduct = this.saveProduct.bind(this)
  }

  componentDidUpdate(prevProps) {
    if (this.props.selected.id !== prevProps.selected.id) {
      const {img, name, price, id} = this.props.selected;

      console.log('prevProps: ', prevProps, 'id: ', this.props.selected)
      this.setState({img, name, price, id, edit: true});
    }
  }

  handleInput = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  cancel = () => {
    this.setState({
      img: '', name: '', price: 0
    })
  }

  saveProduct () {
    axios({
      method: 'post',
      url: '/api/product',
      data: this.state,
      validateStatus: (status) => {
        return true; // I'm always returning true, you may want to do it depending on the status received
      },
    }).catch(error => {
        console.log(error.message)
    }).then(response => {
        // this is now called!
        this.props.invokeGetInventory();
        this.cancel();
    });
  }

  render() {
    return (
      <div className='form'>
        <img alt='' className='form__image' />
        <p className='form__p'>Image URL:</p>
        <input className='form__input' value={this.state.img} onChange={this.handleInput} name='img' />
        <p className='form__p'>Product Name:</p>
        <input className='form__input' value={this.state.name} onChange={this.handleInput} name='name' />
        <p className='form__p'>Price</p>
        <input className='form__input' value={this.state.price} onChange={this.handleInput} name='price' type='number' />
        <div className='form__button'>
          <button className='form__button__cancel' onClick={this.cancel}>Cancel</button>
      {
        this.state.edit === 'true'
        ?
        <button className='form__button__add' onClick={this.saveProduct}>Save changes</button>
        :
        <button className='form__button__add' onClick={this.saveProduct}>Add Inventory</button>
      }
      </div>
      </div>
    )
  }
}

export default Form;
