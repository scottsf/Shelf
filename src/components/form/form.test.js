import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form.js';

let wrapper;

beforeEach(() => {
  const defaultProps = {
    id: 144,
    name: 'TEST',
    price: 10000000,
    img: 'http://test.com',
    match: {
      params: { id: 144 }}
  }
  wrapper = shallow(<Form {...defaultProps}/>);
})

it('Renders Form component without crashing', () => {
  expect(wrapper).toMatchSnapshot()
})




