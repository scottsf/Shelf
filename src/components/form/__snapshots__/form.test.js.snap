// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Form component without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form
    id={144}
    img="http://test.com"
    match={
      Object {
        "params": Object {
          "id": 144,
        },
      }
    }
    name="TEST"
    price={10000000}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt=""
          className="form__image"
        />,
        <p
          className="form__p"
        >
          Image URL:
        </p>,
        <input
          className="form__input"
          name="img"
          onChange={[Function]}
          value=""
        />,
        <p
          className="form__p"
        >
          Product Name:
        </p>,
        <input
          className="form__input"
          name="name"
          onChange={[Function]}
          value=""
        />,
        <p
          className="form__p"
        >
          Price
        </p>,
        <input
          className="form__input"
          name="price"
          onChange={[Function]}
          type="number"
          value=""
        />,
        <div
          className="form__button"
        >
          <Link
            replace={false}
            to="/"
          >
            <button
              className="form__button__cancel"
              onClick={[Function]}
            >
              Cancel
            </button>
          </Link>
          <Link
            replace={false}
            to="/"
          >
            <button
              className="form__button__add"
              onClick={[Function]}
            >
              Add Inventory
            </button>
          </Link>
        </div>,
      ],
      "className": "form",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "className": "form__image",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Image URL:",
          "className": "form__p",
        },
        "ref": null,
        "rendered": "Image URL:",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form__input",
          "name": "img",
          "onChange": [Function],
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Product Name:",
          "className": "form__p",
        },
        "ref": null,
        "rendered": "Product Name:",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form__input",
          "name": "name",
          "onChange": [Function],
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Price",
          "className": "form__p",
        },
        "ref": null,
        "rendered": "Price",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form__input",
          "name": "price",
          "onChange": [Function],
          "type": "number",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              replace={false}
              to="/"
            >
              <button
                className="form__button__cancel"
                onClick={[Function]}
              >
                Cancel
              </button>
            </Link>,
            <Link
              replace={false}
              to="/"
            >
              <button
                className="form__button__add"
                onClick={[Function]}
              >
                Add Inventory
              </button>
            </Link>,
          ],
          "className": "form__button",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="form__button__cancel"
                onClick={[Function]}
              >
                Cancel
              </button>,
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Cancel",
                "className": "form__button__cancel",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Cancel",
              "type": "button",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="form__button__add"
                onClick={[Function]}
              >
                Add Inventory
              </button>,
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add Inventory",
                "className": "form__button__add",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Add Inventory",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt=""
            className="form__image"
          />,
          <p
            className="form__p"
          >
            Image URL:
          </p>,
          <input
            className="form__input"
            name="img"
            onChange={[Function]}
            value=""
          />,
          <p
            className="form__p"
          >
            Product Name:
          </p>,
          <input
            className="form__input"
            name="name"
            onChange={[Function]}
            value=""
          />,
          <p
            className="form__p"
          >
            Price
          </p>,
          <input
            className="form__input"
            name="price"
            onChange={[Function]}
            type="number"
            value=""
          />,
          <div
            className="form__button"
          >
            <Link
              replace={false}
              to="/"
            >
              <button
                className="form__button__cancel"
                onClick={[Function]}
              >
                Cancel
              </button>
            </Link>
            <Link
              replace={false}
              to="/"
            >
              <button
                className="form__button__add"
                onClick={[Function]}
              >
                Add Inventory
              </button>
            </Link>
          </div>,
        ],
        "className": "form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "className": "form__image",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Image URL:",
            "className": "form__p",
          },
          "ref": null,
          "rendered": "Image URL:",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form__input",
            "name": "img",
            "onChange": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Product Name:",
            "className": "form__p",
          },
          "ref": null,
          "rendered": "Product Name:",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form__input",
            "name": "name",
            "onChange": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Price",
            "className": "form__p",
          },
          "ref": null,
          "rendered": "Price",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form__input",
            "name": "price",
            "onChange": [Function],
            "type": "number",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to="/"
              >
                <button
                  className="form__button__cancel"
                  onClick={[Function]}
                >
                  Cancel
                </button>
              </Link>,
              <Link
                replace={false}
                to="/"
              >
                <button
                  className="form__button__add"
                  onClick={[Function]}
                >
                  Add Inventory
                </button>
              </Link>,
            ],
            "className": "form__button",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="form__button__cancel"
                  onClick={[Function]}
                >
                  Cancel
                </button>,
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Cancel",
                  "className": "form__button__cancel",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Cancel",
                "type": "button",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="form__button__add"
                  onClick={[Function]}
                >
                  Add Inventory
                </button>,
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Add Inventory",
                  "className": "form__button__add",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Add Inventory",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
